ironfish/src/rpc/clients/tcpClient.test.ts

```diff
@@ -0,0 +1,125 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+/* eslint-disable @typescript-eslint/ban-ts-comment */
+import { YupUtils } from '../../utils'
+import { IncomingNodeIpcSchema, ValidationError } from '../adapters'
+import { IronfishTcpClient } from './tcpClient'
+
+jest.mock('net')
+
+describe('IronfishTcpClient', () => {
+  const testHost = 'testhost'
+  const testPort = 1234
+  const client: IronfishTcpClient = new IronfishTcpClient(testHost, testPort)
+
+  afterEach(() => {
+    jest.resetAllMocks()
+  })
+
+  it('should connect and disconnect', () => {
+    void client.connect()
+    expect(client.client.connect).toHaveBeenCalledWith(testPort, testHost)
+
+    client.close()
+    expect(client.client.end).toHaveBeenCalled()
+  })
+
+  it('should send messages in the node-ipc encoding', async () => {
+    const messageId = 1
+    const route = 'foo/bar'
+
+    const expectedMessage =
+      JSON.stringify({
+        type: 'message',
+        data: {
+          mid: messageId,
+          type: route,
+        },
+      }) + '\f'
+
+    const result = await YupUtils.tryValidate(IncomingNodeIpcSchema, expectedMessage.trim())
+    expect(result.error).toBe(null)
+
+    client.request(route)
+    expect(client.client.write).toHaveBeenLastCalledWith(expectedMessage)
+  })
+
+  it('should handle message responses', async () => {
+    expect.assertions(1)
+
+    // @ts-ignore
+    const spyOnMessage = jest.spyOn(client, 'onMessage')
```

Comment:
Can we just test the TcpClient.request method here instead of testing the internal (private) methods?